pkg/cloud/azure/actuators/machine/actuator_test.go

```diff
@@ -521,6 +523,59 @@ func TestAvailabilityZones(t *testing.T) {
 	}
 }
 
+func TestGetZone(t *testing.T) {
+	testCases := []struct {
+		inputZone *string
+		expected  string
+	}{
+		{
+			inputZone: nil,
+			expected:  "",
+		},
+		{
+			inputZone: pointer.StringPtr("3"),
+			expected:  "3",
+		},
+		{
+			inputZone: pointer.StringPtr(""),
+			expected:  "random",
+		},
+	}
+
+	for _, tc := range testCases {
+		fakeScope := newFakeScope(t, v1alpha1.ControlPlane)
+		fakeReconciler := newFakeReconcilerWithScope(t, fakeScope)
+		fakeReconciler.scope.MachineConfig.Zone = tc.inputZone
+
+		zones := []string{"1", "2", "3"}
+		fakeReconciler.availabilityZonesSvc = &FakeAvailabilityZonesService{
+			zonesResponse: zones,
+		}
+
+		got, err := fakeReconciler.getZone(context.Background())
+		if err != nil {
+			t.Errorf("unexpected error getting zone")
```

Comment:
Can the remainder legitimately continue if this errors?

Reply:
no. Thanks!